src/random.h

```diff
@@ -9,6 +9,7 @@
 #include "crypto/chacha20.h"
 #include "crypto/common.h"
 #include "uint256.h"
+#include "compat.h"
```

Comment:
Please just make NUM_OS_RANDOM_BYTES an int to keep the compat.h include out of this header. I realize it's still needed in the cpp, but let's try not to spread the use further than necessary.